File: Hosts/Python3/ToGo/sanity/Main_test.py
import unittest
from TestFuncs_MetaFFIHost import *
import collections

metaffi_load('TestFuncs_MetaFFIGuest')
class TestSanity(unittest.TestCase):

	def test_hello_world(self):
		HelloWorld()
	

	def test_returns_an_error(self):
		try:
			ReturnsAnError()
			self.fail('Test should have failed')
		except:
			pass

	
	def test_div_integers(self):
		res = DivIntegers(1, 2)
		if res != 0.5:
			self.fail('Expected 0.5, got: '+str(res))

		try:
			DivIntegers(1, 0)
			self.fail('Expected an error - divisor is 0')
		except:
			pass

	
	def test_join_strings(self):
		res = JoinStrings(['A','b','C'])
		if res != 'A,b,C':
			self.fail('Expected A,b,C. Got: '+res)

	def test_wait_a_bit(self):
		fivesec = GetFiveSeconds_metaffi_getter()
		WaitABit(fivesec)

	def test_test_map(self):
		map = TestMap(NewTestMap())

		map.Set('x', 250)
		if not map.Contains('x'):
			self.fail('Map should contain x')

		if map.Get('x') != 250:
			self.fail('x should be 250')

		map.Set('y', 'test')
		if not map.Contains('y'):
			self.fail('Map should contain y')

		if map.Get('y') != 'test':
			self.fail('y should be \'test\'')

		deq = collections.deque()
		deq.append(600)
		map.Set('z', deq)
		if not map.Contains('z'):
			self.fail('Map should contain z')

		mapped_deq = map.Get('z')
		val = mapped_deq.pop()
		if val != 600:
			self.fail('mapped_deq should contain 600')

		map.SetName_metaffi_setter('MyName')
		newname = map.GetName_metaffi_getter()
		if newname != 'MyName':
			self.fail('TestMap.Name should be MyName and it is '+newname)
		
		
		

File: Hosts/Go/ToGo/sanity/Main_test.go
//+build maintest

package main

import "testing"

//--------------------------------------------------------------------
func TestHelloWorld(t *testing.T){
	err := HelloWorld()
	if err != nil{
		t.Fatal(err)
	}
}
//--------------------------------------------------------------------
func TestReturnsAnError(t *testing.T){
	err := ReturnsAnError()
	if err == nil{
		t.Fatal("Error expected")
	}
}
//--------------------------------------------------------------------
func TestDivIntegers(t *testing.T){
	res, err := DivIntegers(1, 2)
	if err != nil{
		t.Fatal(err)
	}

	if res != 0.5{
		t.Fatalf("Expected 0.5, got: %v", res)
	}

	res, err = DivIntegers(1, 0)
	if err == nil{
		t.Fatal("Expected an error - divisor is 0")
	}
}
//--------------------------------------------------------------------
func TestJoinStrings(t *testing.T){
	res, err := JoinStrings([]string{"A", "b", "C"})
	if err != nil{
		t.Fatal(err)
	}

	if res != "A,b,C"{
		t.Fatalf("Expected A,b,C. Got: %v", res)
	}
}
//--------------------------------------------------------------------
func TestTestMap(t* testing.T){

	m, err := NewTestMap()
	if err != nil{ t.Fatal(err) }
	
	err = SetKey(m, "k1", "v1")
	if err != nil{ t.Fatal(err) }

	exists, err := ContainsKey(m, "k1")
	if err != nil{ t.Fatal(err) }
	if !exists{ t.Fatalf("k1 should exist in map") }

	v, err := GetKey(m, "k1")
	if err != nil{ t.Fatal(err) }

	if v != "v1"{ t.Fatalf("k1 value should be v1, while it is: %v", v) }
}
//--------------------------------------------------------------------